core/trino-main/src/main/java/io/trino/sql/planner/QueryPlanner.java

```diff
@@ -367,6 +363,7 @@ protected PlanNode visitPlan(PlanNode node, RewriteContext<Void> context)
     public RelationPlan plan(QuerySpecification node)
     {
         PlanBuilder builder = planFrom(node);
+        builder = planWindowDefinitions(builder, analysis.getWindowDefinitions(node), node);
```

Comment:
I found out this is not needed at all.

Because all the window specifications of WINDOW clause were analyzed in the right scope (which is the FROM scope), then thanks to TranslationMap they should be planned properly (that is, using source symbols) no matter where they appear. There's no need to plan them "in advance" at the bottom of the plan.

Lazy planning will also solve the issue of unnecessarily planning window specifications which are never referenced.